--- view.php	2012-08-01 10:56:34.357671620 +0500
+++ view.php	2012-08-08 08:08:25.000000000 +0500
@@ -3,10 +3,9 @@
 //  Display the course home page.
 
     require_once('../config.php');
-    require_once('lib.php');
-    require_once($CFG->dirroot.'/mod/forum/lib.php');
-    require_once($CFG->libdir.'/completionlib.php');
-    require_once('../mod/attforblock/locallib.php');
+    require_once($CFG->dirroot.'/mod/assignment/lib.php');//Added By Hina Yousuf
+ 	require_once($CFG->libdir.'/filelib.php');//hina
+    require_once("$CFG->libdir/filestorage/stored_file.php");
     $id          = optional_param('id', 0, PARAM_INT);
     $name        = optional_param('name', '', PARAM_RAW);
     $edit        = optional_param('edit', -1, PARAM_BOOL);
@@ -17,6 +16,7 @@
     $move        = optional_param('move', 0, PARAM_INT);
     $marker      = optional_param('marker',-1 , PARAM_INT);
     $switchrole  = optional_param('switchrole',-1, PARAM_INT);
+    $download = optional_param('download',false, PARAM_BOOL);
 
     if (empty($id) && empty($name) && empty($idnumber)) {
         print_error('unspecifycourseid', 'error');
@@ -49,6 +49,18 @@
     }
 
     require_login($course);
+    //Added By Hina Yousuf
+        if($download){
+                if(has_capability('moodle/course:downloadmaterial', $context)) {
+                         course_material($course);
+                }
+        }
+ 	require_once('lib.php');
+    require_once($CFG->dirroot.'/mod/forum/lib.php');
+    require_once($CFG->libdir.'/completionlib.php');
+    require_once('../mod/attforblock/locallib.php');
+    require_once($CFG->dirroot.'/mod/assignment/lib.php');//Added By Hina Yousuf
+    
 
     // Switchrole - sanity check in cost-order...
     $reset_user_allowed_editing = false;
@@ -455,6 +467,14 @@ $sql="select parent,name from mdl_course
         }
 
         ////end of feedback not given message
+         ///Download Course material
+                //Added By Hina Yousuf
+                if(has_capability('moodle/course:downloadmaterial', $context)) {
+                        echo                    '<form method="post" style="display: inline; margin: 0; padding: 0;">';
+                        echo                    '<input type="hidden" name="download" value="true" /><input type="submit" value="Download Course Material" /></form>';
+                }
+                //end
+        
 
 	$modinfo =& get_fast_modinfo($COURSE);
     get_all_mods($course->id, $mods, $modnames, $modnamesplural, $modnamesused);
@@ -499,4 +519,3 @@ $sql="select parent,name from mdl_course
 
     echo $OUTPUT->footer();
 
-
